### ISSUE TITLE
Docker image build fails test for perl<5.20 on CentOS 7.1 with SELinux=enforcing

### ISSUE THREAD

**Issue by rubensteintech** [Perl/docker-perl#17]
Docker builds of Perl less than 5.20 on CentOS 7.1.1503 when SELinux is set to "enforcing" (SELINUX=enforcing) tests fail with these errors during test:

```
$ getenforce 
Enforcing
$ cd docker-perl/5.010.001-64bit/
$ docker build -t perl:5.10 .

...

#   Failed test 'queue a message'
#   at ../ext/IPC-SysV/t/ipcsysv.t line 165.
# The failure of the subtest #2 may indicate that the message queue
# resource limits either of the system or of the testing account
# have been reached.  Error message "Operating would block" is
# usually indicative of this situation.  The error message was now:
# "Permission denied"
#
# You can check the message queues with the 'ipcs' command and
# you can remove unneeded queues with the 'ipcrm -q id' command.
# You may also consider configuring your system or account
# to have more message queue resources.
#
# Because of the subtest #2 failing also the substests #5 and #6 will
# very probably also fail.

#   Failed test 'message get call'
#   at ../ext/IPC-SysV/t/ipcsysv.t line 196.
# This failure was to be expected because the subtest #2 failed.
#   Failed test 'message get data'
#   at ../ext/IPC-SysV/t/ipcsysv.t line 214.
# Looks like you failed 3 tests of 38.
../ext/IPC-SysV/t/ipcsysv.t ......................................
```

The issue appears to be with the version of the IPC::SysV module included in the Perl distribution, as the latest version of IPC::SysV (via CPAN) does compile and pass tests independently with SELINUX=enforcing.

Docker builds for 5.16 and 5.10 have been confirmed to run successfully on CentOS 7.1 when SELinux is disabled (SELINUX=disabled).

**Comment by sobrique** [comment]
As a workaround, I have been using: `setenforce` - this allows you to `setenforce Permissive` before running the tests, and `setenforce Enforcing` to set it back again. This allows you to bypass the tests that fail just as a result of selinux (of course, not guaranteeing they'll actually work when you run the module with `enforcing` enabled)

**Comment by zakame** [comment]
Just re-checked this (sorry for the long late reply.)

For modern Docker (currently version 17.06) installation on CentOS 7 now pulls in the `container-selinux` package from Project Atomic:

```
[root@localhost ~]# yum info container-selinux
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.rise.ph
 * extras: mirror.rise.ph
 * updates: mirror.rise.ph
Installed Packages
Name        : container-selinux
Arch        : noarch
Epoch       : 2
Version     : 2.19
Release     : 2.1.el7
Size        : 33 k
Repo        : installed
From repo   : extras
Summary     : SELinux policies for container runtimes
URL         : https://github.com/projectatomic/container-selinux
License     : GPLv2
Description : SELinux policy modules for use with container runtimes.
```

There are now other issues (mainly related to base image updates/churn) that prevent Perl < 5.20 to build, see #34.  I'll close this issue once we've also fixed #34.

**Comment by zakame** [comment]
Latest release (tagged `r2010817.0` as of today) now includes #35 and published to the Hub.  Closing this for now.  Thanks!
